src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3159,7 +3159,7 @@ internal Buffer(IEnumerable<TElement> source, bool queryInterfaces = true)
 
         internal TElement[] ToArray()
         {
-            if (count == 0) return new TElement[0];
+            if (count == 0) return Array.Empty<TElement>();
```

Comment:
Great idea; I agree, many of the places where this is used (`Iterator.ToArray`) seem to imply that they are returning a different array. I'll revert and add a comment on the side.